app/src/ingestion/process_crawl_jobs.py

```diff
@@ -1,10 +1,268 @@
 "Process scheduled crawl jobs by extracting support resources from domains using OpenAI web search."
 
+import asyncio
+import json
 import logging
+from typing import Iterable
+
+from haystack import AsyncPipeline
+
+from src.adapters import db
+from src.app_config import config
+from src.common import phoenix_utils
+from src.common.components import OpenAIWebSearchGenerator
+from src.db.models.crawl_job import CrawlJob
+from src.db.models.support_listing import Support, SupportListing
+from src.ingestion.support_entry import SUPPORT_ENTRY_SCHEMA, SupportEntry
+from src.util import datetime_util
 
 logger = logging.getLogger(__name__)
 
 
+def create_websearch() -> OpenAIWebSearchGenerator:
+    return OpenAIWebSearchGenerator()
+
+
+def build_pipeline() -> AsyncPipeline:
+    """
+    Build a Haystack pipeline for processing crawl jobs.
+
+    Args:
+        prompt_name: Name of the Phoenix prompt to use
+
+    Returns:
+        Configured AsyncPipeline
+    """
+    pipe = AsyncPipeline()
+    pipe.add_component("llm", create_websearch())
+    return pipe
+
+
+async def run_pipeline(pipeline: AsyncPipeline, job: CrawlJob) -> list[dict]:
+    """
+    Run the pipeline for a single crawl job.
+
+    Args:
+        pipeline: The Haystack AsyncPipeline to run
+        job: The CrawlJob to process
+
+    Returns:
+        List of support entry dictionaries
+    """
+    logger.info("Running pipeline for job: domain=%s, prompt=%s", job.domain, job.prompt_name)
+
+    # We are setting the prompt at runtime, so can't use ChatPromptBuilder here
```

Comment:
It should be possible to use `ChatPromptBuilder` with a dynamic template -- see https://github.com/navapbc/labs-referral-pilot/blob/e0f0e8fb20955f7963984b0ac970ed48ba3fe8d6/app/src/pipelines/generate_referrals/pipeline_wrapper.py/#L100-L101